### ISSUE TITLE
Inconsistent git hashes for sub-pkgs of a single repo cause trouble with no warning

### ISSUE THREAD

**Issue by thockin** [tools/godep#223]
We accidentally had one sub-pkg of an upstream repo where the githash was different than the rest.  Godep checked out this (older) hash and obliterated files that we needed by other sub-pkgs.  I feel like there should be a warning, something like "different revs for sub-packages of the same repo - this is not expected and probably bad things will happen"

**Comment by kr** [comment]
The save and update commands check for this case and prevent it from happening. Godeps.json should never contain two different commit hashes from the same repo. If it happens anyway, that's a bug.

**Comment by thockin** [comment]
I'm not sure how it happened, but it did - hence my suggestion that maybe it should yell at me.

**Comment by freeformz** [comment]
@thockin Can you re-produce on a public repo that I can test with / base a test on?

**Comment by thockin** [comment]
I don't know HOW it happened, so I can't reproduce it.  Obviously I can
manually recreate it by having a Godeps file with two pkgs of teh same git
repo at different hashes - you don't need me to set that up (but I will if
you want :)

**Comment by freeformz** [comment]
I assume this happened on a `godep restore`?

TBH: I'm more interested in how you got into that situation in the first case, which is what I'd like a reproducer for.

**Comment by thockin** [comment]
yeah, I have no idea how it happened - it may well have been a hand edit.
The issue was that it took FOREVER to figure out what was going wrong
because the tool knew but didn't offer any clues.

**Comment by kr** [comment]
@thockin is right tho, even though godep tries to avoid this situation, it's reasonable to also check for it later, because who knows what happened to the file in between.

**Comment by freeformz** [comment]
@thockin @kr agreed.

**Comment by freeformz** [comment]
Closing issues before archiving. Thank You\!
